### PR TITLE
Release `v4.0.0 alpha.2`

### PR DESCRIPTION
**Pull request by ascjones**
#### New `ink` crate
The `ink_lang` crate has been replaced in [#1223](https://github.com/paritytech/ink/pull/1223) by a new top level `ink`
crate. All existing sub-crates are reexported and should be used via the new `ink` crate, so e.g. `ink::env` instead of
`ink_env`. Contract authors should now import the top level `ink` crate instead of the individual crates.

##### Migration
- In `Cargo.toml` Replace all individual `ink_*` crate dependencies with the `ink` crate.
- In the contract source:
  - Remove the commonly used `use ink_lang as ink` idiom.
  - Replace all usages of individual crates with reexports, e.g. `ink_env` ➜ `ink::env`.

#### Storage Rework
[#1331](https://github.com/paritytech/ink/pull/1331) changes the way `ink!` works with contract storage. Storage keys 
are generated at compile-time, and user facing abstractions which determine how contract data is laid out in storage
have changed.

##### Migration
- Initialize `Mapping` fields with `Mapping::default()` instead of  `ink_lang::utils::initialize_contract` in
constructors. See [`erc20`](./examples/erc20/lib.rs) and other examples which use a `Mapping`.
- Remove `SpreadAllocate`, `SpreadLayout` and `PackedLayout` implementations.

#### Removal of `wee-alloc` support
ink! uses a bump allocator by default, additionally we supported another allocator (`wee-alloc`)
through a feature flag. `wee-alloc` is no longer maintained and we removed support for it.

### Changed
- Introduce `ink` entrance crate ‒ [#1223](https://github.com/paritytech/ink/pull/1223)
- Use `XXH32` instead of `sha256` for calculating storage keys ‒ [#1393](https://github.com/paritytech/ink/pull/1393)

### Fixed
- Trim single whitespace prefix in the metadata `docs` field ‒ [#1385](https://github.com/paritytech/ink/pull/1385)
- Allow pay_with_call to take multiple arguments ‒ [#1401](https://github.com/paritytech/ink/pull/1401)

### Added
- Add `ink_env::pay_with_call!` helper macro for off-chain emulation of sending payments with contract message calls ‒ [#1379](https://github.com/paritytech/ink/pull/1379)

### Removed
- Remove `wee-alloc` ‒ [#1403](https://github.com/paritytech/ink/pull/1403)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## Version 4.0.0-alpha.1
```
Comment (cmichi) [inline CHANGELOG.md:9]:
```suggestion
## Version 4.0.0-alpha.2
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -30,6 +42,9 @@ through a feature flag. `wee-alloc` is no longer maintained and we removed suppo
 ### Fixed
 - Trim single whitespace prefix in the metadata `docs` field ‒ [#1385](https://github.com/paritytech/ink/pull/1385)
 
+### Added
+- Add ink_env::pay_with_call! helper macro for off-chain emulation of sending payments with contract msg calls - [#1379](https://github.com/paritytech/ink/pull/1379)
```
Comment (cmichi) [inline CHANGELOG.md:46]:
```suggestion
- Add `ink_env::pay_with_call!` helper macro for off-chain emulation of sending payments with contract msg calls ‒ [#1379](https://github.com/paritytech/ink/pull/1379)
```

**Review by cmichi — APPROVED** [review]
Please include the suggestions I made, other than that 👍 .

**Comment by HCastano** [comment]
Small thing, can we make sure #1401 gets in to the release?

**Comment by ascjones** [comment]
> Small thing, can we make sure #1401 gets in to the release?

Sure, I will pick this up again tomorrow and include that

**Review by HCastano — APPROVED** [review]
:shipit:

**Pull request merged**